public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -45,7 +46,10 @@ interface CacheValue {
 
 // this little monster combines prometheus rules and ruler rules to produce a unified data structure
 // can limit to a single rules source
-export function useCombinedRuleNamespaces(rulesSourceName?: string): CombinedRuleNamespace[] {
+export function useCombinedRuleNamespaces(
+  rulesSourceName?: string,
+  promRuleNamespaces?: RuleNamespace[]
```

Comment:
```suggestion
  grafanaPromRuleNamespaces?: RuleNamespace[]
```
I think with this name is more clear that we are only optionally passing Grafana promrules

Reply:
I'm a bit confused about this, aren't we also using `useCombinedRuleNamespaces` for prometheus (non-Grafana managed) too?

Reply:
yes, but this parameter is only used in case of grafana alert data source

Reply:
Yeap, we use it for all other prometheus rules as well. The new parameter contains only grafana-managed though, which are the ones that present the bug as those are filtered by the backend.

Reply:
Gotcha, that makes sense! 👍